src/Umbraco.Core/IO/FileSystems.cs

```diff
@@ -305,7 +305,8 @@ public TFileSystem GetFileSystemProvider<TFileSystem>(Func<IFileSystem> fallback
 
                 // fixme - switch to using container. where are these registered?
 
-                var fs = (IFileSystem) Activator.CreateInstance(typeof(TFileSystem), shadowWrapper);
+                //var fs = (IFileSystem) Activator.CreateInstance(typeof(TFileSystem), shadowWrapper);
+                var fs = Current.Container.GetInstance<TFileSystem>((IFileSystem)shadowWrapper);
```

Comment:
@Shazwazza seems like you've done something like this before: https://github.com/seesharper/LightInject/issues/237

Got any idea why it complains it can't resolve the dependency? We get this down, we've basically got all tight couplings solved.

See ContainerAdapter.cs line 48 for concrete impl. Basic forwarding of object array to concrete container. (No extension usage)

Reply:
nvm, got the hang of it.